### PR TITLE
Fix bug that assumes all libraries have at least two packages

### PR DESCRIPTION
**Pull request by andrie**
This is a fix for the bug originally reported at https://support.rstudio.com/hc/communities/public/questions/203052576-setting-libpath-in-Rstudio

To replicate the error, use:

```r
loc <- normalizePath(tempdir())
assign(".lib.loc", loc, envir = environment(.libPaths))
```

The resulting error message is:

```
Error in pkgs.name[[i]] : subscript out of bounds
```

This happens because the code simplifies the result of `installed.packages()`, and if the library contains zero or one row, the simplification results in a vector, not a data frame.

The fix is to set `drop=FALSE` when returning columns of the data frame.

This bug manifests in the [`checkpoint` package on CRAN](https://cran.r-project.org/package=checkpoint). In the package is a workaround to fix this bug, and this has been working without problem for the best part of two years.  See the [checkpoint repo](https://github.com/RevolutionAnalytics/checkpoint/blob/master/R/fixRstudioBug.R).

### CODE REVIEW

**Comment by jjallaire** [comment]
@kevinushey This LGTM. Could you review and merge?

File: src/cpp/session/modules/SessionPackages.R [hunk 1]
```diff
@@ -315,7 +315,7 @@ if (identical(as.character(Sys.info()["sysname"]), "Darwin") &&
                          desc=pkgs.desc,
                          url=pkgs.url,
                          loaded=pkgs.loaded,
-                         check.rows = TRUE,
+                         check.rows = FALSE,
```
Comment (kevinushey) [inline src/cpp/session/modules/SessionPackages.R]: What's the motivation for this particular change?

Comment (andrie) [inline src/cpp/session/modules/SessionPackages.R]: I'll be honest that I don't understand why this is necessary. This argument checks that row names are consistent when creating a `data.frame`, and something seems to go wrong when checking row names on an empty data frame.

But I do know that I could not get this function to work in any other way.

**Comment by kevinushey** [comment]
FWIW I still see the bug with your reproducible example (thanks for providing one!)

```
Error in pkgs.name[[i]] : subscript out of bounds
```

and I think it's because this code here:

https://github.com/rstudio/rstudio/blob/538e0e4eddf3c92849f8c27df699d1410fc68a1a/src/cpp/session/modules/SessionPackages.R#L302

assumes that there is at least one package in the specified library paths, whereas the provided test case simulates a library path with no packages. I think that implies that this PR needs a few more changes before we can merge. (we should probably use `seq_along()` in the for loop there)

**Comment by andrie** [comment]
Agreed. I have a fix. Coming your way shortly.

**Comment by andrie** [comment]
So the fix is to replace that for loop (ugh!) with `sapply()`:

```r
pkgs.version <- sapply(seq_along(pkgs.name), function(i){
        .rs.packageVersion(pkgs.name[[i]], pkgs.library[[i]], instPkgs)
      })
```

And now the `check.rows=FALSE` is no longer necessary, so I reverted that.

**Comment by kevinushey** [comment]
Awesome, thanks for that! LGTM.

**Pull request merged**